sdk/search/Azure.AI.Search/src/SearchClientOptions.cs

```diff
@@ -0,0 +1,173 @@
+            /// <summary>
+            /// The 2019-05-06 version of the service that is Generally Available.
+            /// </summary>
+            V2019_05_06 = 1
+            #pragma warning restore CA1707
+        }
+
+        /// <summary>
+        /// The Latest service version supported by this client library.
+        /// </summary>
+        internal const ServiceVersion LatestVersion = ServiceVersion.V2019_05_06;
+
+        /// <summary>
+        /// Gets the <see cref="ServiceVersion"/> of the service API used when
+        /// making requests.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/azure/search/search-api-versions" />.
+        /// </summary>
+        public ServiceVersion Version { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchClientOptions"/>
+        /// class.
+        /// </summary>
+        /// <param name="version">
+        /// An optional <see cref="ServiceVersion"/> to specify the version of
+        /// the REST API to use.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/azure/search/search-api-versions" />.
+        ///
+        /// If not provided, the <paramref name="version"/> will default to the
+        /// latest supported by this client library.  It is recommended that
+        /// application authors allow the version to float to the latest and
+        /// library authors pin to a specific version.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the <paramref name="version"/> is not supported by this
+        /// client library.
+        /// </exception>
+        public SearchClientOptions(ServiceVersion version = LatestVersion)
+        {
+            Version = version.Validate();
+            AddLoggingHeaders();
+            AddLoggingQueryParameters();
+        }
+
+        /// <summary>
+        /// Create an <see cref="HttpPipeline"/> to send requests to the Search
+        /// Service.
+        /// </summary>
+        /// <param name="credential">
+        /// The <see cref="SearchApiKeyCredential"/> to authenticate requests.
+        /// </param>
+        /// <returns>An <see cref="HttpPipeline"/> to send requests.</returns>
+        internal HttpPipeline Build(SearchApiKeyCredential credential)
+        {
+            Debug.Assert(credential != null);
+            return HttpPipelineBuilder.Build(
+                options: this,
+                perCallPolicies: new[] { new SearchApiKeyCredentialPolicy(credential) },
+                perRetryPolicies: Array.Empty<HttpPipelinePolicy>(),
+                responseClassifier: null);
+        }
+
+        /// <summary>
+        /// Add the allow list headers to the <see cref="DiagnosticsOptions"/>
+        /// that are considered safe for logging/exceptions by default.
+        /// </summary>
+        private void AddLoggingHeaders()
```

Comment:
client-request-id and friends?

Reply:
Yes - at some point (soon-ish) I plan to collect everything out of the swagger and allow-list it.